StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1108BlockStatementsMustNotContainEmbeddedComments.cs

```diff
@@ -58,19 +61,90 @@ public class SA1108BlockStatementsMustNotContainEmbeddedComments : DiagnosticAna
             ImmutableArray.Create(Descriptor);
 
+        private readonly SyntaxKind[] supportedKinds = new[]
+        {
+            SyntaxKind.ForEachStatement,
+            SyntaxKind.ForStatement,
+            SyntaxKind.WhileStatement,
+            SyntaxKind.DoStatement,
+            SyntaxKind.IfStatement,
+            SyntaxKind.ElseClause,
+            SyntaxKind.LockStatement,
+            SyntaxKind.TryStatement,
+            SyntaxKind.CatchClause,
+            SyntaxKind.FinallyClause,
+            SyntaxKind.CheckedStatement,
+            SyntaxKind.UncheckedStatement,
+            SyntaxKind.FixedStatement
+        };
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => SupportedDiagnosticsValue;
+
         /// <inheritdoc/>
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(this.AnalyzeBlock, SyntaxKind.Block);
+            context.RegisterSyntaxNodeAction(this.AnalyzeSwitch, SyntaxKind.SwitchStatement);
+        }
+
+        private void AnalyzeSwitch(SyntaxNodeAnalysisContext context)
         {
-            get
+            var switchStatement = (SwitchStatementSyntax) context.Node;
+            var openBraceToken = switchStatement.OpenBraceToken;
+            if (openBraceToken.IsMissing)
             {
-                return SupportedDiagnosticsValue;
+                return;
             }
+
+            var previousToken = openBraceToken.GetPreviousToken();
+            if (previousToken.IsMissing)
+            {
+                return;
+            }
+
+            this.FindAllComments(context, previousToken, openBraceToken);
         }
 
-        /// <inheritdoc/>
-        public override void Initialize(AnalysisContext context)
+        private void AnalyzeBlock(SyntaxNodeAnalysisContext context)
+        {
+            var block = (BlockSyntax) context.Node;
+            if (block.Parent == null || !this.supportedKinds.Any(block.Parent.IsKind))
+            {
+                return;
+            }
+
+            var openBraceToken = block.OpenBraceToken;
+            if (openBraceToken.IsMissing)
+            {
+                return;
+            }
+
+            var previousToken = openBraceToken.GetPreviousToken();
+            if (previousToken.IsMissing)
+            {
+                return;
+            }
+
+            this.FindAllComments(context, previousToken, openBraceToken);
+        }
+
+        private void FindAllComments(SyntaxNodeAnalysisContext context, SyntaxToken previousToken, SyntaxToken openBraceToken)
+        {
+            var comments = previousToken.TrailingTrivia.Where(this.IsComment)
+                .Union(openBraceToken.LeadingTrivia.Where(this.IsComment))
+                .ToList();
```

Comment:
Because you are only enumerating comments once you can remove ToList(). This saves you some unnessecaray allocations. You can also exchange Union with Concat. The difference between these is that Union removes duplicates which you cant have here anyways.